pkg/api/networkservice/mechanisms/wireguard/helpers.go

```diff
@@ -0,0 +1,130 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wireguard
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+// Mechanism - a wireguard mechanism utility wrapper
+type Mechanism interface {
+	// SrcIP -  src ip
+	SrcIP() net.IP
+	// DstIP - dst ip
+	DstIP() net.IP
+	// SrcPublicKey - source public key
+	SrcPublicKey() (string, error)
+	// DstPublicKey - destination public key
+	DstPublicKey() (string, error)
+	// SrcPort - Source interface listening port
+	SrcPort() (int, error)
+	// SrcPort - Destination interface listening port
+	DstPort() (int, error)
+}
+
+type mechanism struct {
+	*networkservice.Mechanism
+}
+
+// ToMechanism - convert unified mechanism to useful wrapper
+func ToMechanism(m *networkservice.Mechanism) Mechanism {
+	if m.Type == MECHANISM {
+		return &mechanism{
+			m,
+		}
+	}
+	return nil
+}
+
+func (m *mechanism) SrcIP() net.IP {
+	return net.ParseIP(m.GetParameters()[SrcIP])
+}
+
+func (m *mechanism) DstIP() net.IP {
+	return net.ParseIP(m.GetParameters()[DstIP])
+}
+
+func (m *mechanism) stringValue(parameter string) (string, error) {
+	if m == nil {
+		return "", errors.New("mechanism cannot be nil")
+	}
+
+	if m.GetParameters() == nil {
+		return "", errors.Errorf("mechanism.Parameters cannot be nil: %v", m)
+	}
+
+	value, ok := m.Parameters[parameter]
+	if !ok {
+		return "", errors.Errorf("mechanism.Type %s requires mechanism.Parameters[%s]", m.GetType(), parameter)
+	}
+
+	return value, nil
+}
+
+// SrcPublicKey returns the SrcPublicKey parameter of the Mechanism
+func (m *mechanism) SrcPublicKey() (string, error) {
+	return m.stringValue(SrcPublicKey)
+}
+
+// DstPublicKey returns the DstPublicKey parameter of the Mechanism
+func (m *mechanism) DstPublicKey() (string, error) {
+	return m.stringValue(DstPublicKey)
+}
+
+// SrcPort - Source interface listening port
+func (m *mechanism) SrcPort() (int, error) {
+	srcPortStr, err := m.stringValue(SrcPort)
+	if err != nil {
+		return 0, err
+	}
+
+	srcPort, err := strconv.ParseInt(srcPortStr, 10, 64)
+	if err != nil {
+		return 0, errors.Wrapf(err, "cannot parse mechanism.Parameters[%s]=%v value", SrcPort, srcPortStr)
+	}
+
+	return int(srcPort), nil
+}
+
+// DstPort - Destination interface listening port
+func (m *mechanism) DstPort() (int, error) {
+	dstPortStr, err := m.stringValue(DstPort)
+	if err != nil {
+		return 0, err
+	}
+
+	dstPort, err := strconv.ParseInt(dstPortStr, 10, 64)
+	if err != nil {
+		return 0, errors.Wrapf(err, "cannot parse mechanism.Parameters[%s]=%v value", DstPort, dstPortStr)
+	}
+
+	return int(dstPort), nil
+}
+
+// AssignPort - generate unique port by connection ID for wireguard connection
+func AssignPort(connID string) string {
```

Comment:
Question... is this simply a setter?  If so, would it make sense to call it SetPort?

Reply:
thanks, should be fixed.